### PR TITLE
More fixes for the release

### PR DESCRIPTION
**Pull request by adithyaov**

### CODE REVIEW

File: benchmark/Streamly/Benchmark/Prelude/WSerial.hs [hunk 1]
```diff
@@ -193,9 +193,13 @@ o_1_space_concat value =
             "concatMapWithInterleave (n of 1)"
             (concatStreamsWith Internal.interleave value 1)
         -}
+
+        -- XXX stack overflow with --long
```
Comment (harendra-kumar) [inline benchmark/Streamly/Benchmark/Prelude/WSerial.hs:197]: Was this working in previous release?

Comment (harendra-kumar) [reply benchmark/Streamly/Benchmark/Prelude/WSerial.hs:197]: interleave and roundrobin are unlikely to work with large number of streams, definitely not with infinite number of streams, therefore they should be re-classified to o-n-space. I am wondering how it was passing in earlier releases? We also have this comment above:

-- concatMapWith using StreamD versions of interleave operations are
-- all quadratic, we just measure the sqrtVal benchmark for comparison.

Comment (adithyaov) [reply benchmark/Streamly/Benchmark/Prelude/WSerial.hs:197]: They were not working with the previous release either.

File: Changelog.md [hunk 2]
```diff
@@ -19,6 +19,15 @@
 * Fix array: missing pointer touch could potentially cause use of freed memory.
 * Fix array: unnecessary additional allocation due to a bug
 
+### Notable Internal API Changes
+
+* `tapAsync` is moved from `Stream.Parallal` to `Stream.IsStream.Transform`. It
```
Comment (harendra-kumar) [inline Changelog.md:24]: typo `Parallal`. `tapAsync` was not moved, it was already present, and it already took a Fold. We can remove this bullet.

File: Changelog.md [hunk 3]
```diff
@@ -19,6 +19,15 @@
...
+  now takes a `Fold` intead of a reducing function.
+* `tapAsyncK` has been added in `Stream.IsStream.Transform` preserving the
+  original behaviour of `tapAsync`.
```
Comment (harendra-kumar) [inline Changelog.md:27]: `tapAsyncK` has been moved from `Streamly.Internal.Data.Stream.Parallel` and renamed, it was called `tapAsync` in the original module.

File: Changelog.md [hunk 4]
```diff
@@ -19,6 +19,11 @@
 * Fix array: missing pointer touch could potentially cause use of freed memory.
 * Fix array: unnecessary additional allocation due to a bug
 
+### Notable Internal API Changes
+
+* `tapAsyncK` has been moved from `Streamly.Internal.Data.Stream.Parallel` and
+  renamed, it was called `tapAsync` in the original module.
```
Comment (harendra-kumar) [inline Changelog.md:25]: `tapAsync` from `Streamly.Internal.Data.Stream.Parallel` has been moved to `Streamly.Internal.Data.Stream.IsStream` and renamed to `tapAsyncK`.

**Review by harendra-kumar — APPROVED** [review]

**Pull request merged**